packages/cli/src/modules/breaking-changes/types/rule.types.ts

```diff
@@ -0,0 +1,78 @@
+import type { WorkflowEntity } from '@n8n/db';
+
+export const enum BreakingChangeSeverity {
+	CRITICAL = 'critical',
+	HIGH = 'high',
+	MEDIUM = 'medium',
+	LOW = 'low',
+}
+
+export const enum BreakingChangeCategory {
+	WORKFLOW = 'workflow',
+	INSTANCE = 'instance',
+	ENVIRONMENT = 'environment',
+	DATABASE = 'database',
+	INFRASTRUCTURE = 'infrastructure',
+}
+
+export const enum IssueLevel {
+	ERROR = 'error',
+	WARNING = 'warning',
+	INFO = 'info',
+}
+
+export type BreakingChangeVersion = 'v2';
+
+export interface BreakingChangeMetadata {
+	id: string;
+	version: BreakingChangeVersion;
+	title: string;
+	description: string;
+	category: BreakingChangeCategory;
+	severity: BreakingChangeSeverity;
+	documentationUrl?: string;
+}
+
+export interface WorkflowIssue {
```

Comment:
What do you think about merging `WorkflowIssue` and `InstanceIssue` into one `DetectedIssue` interface?